crates/grafbase-sdk/src/test/mock.rs

```diff
@@ -0,0 +1,62 @@
+mod builder;
+mod entity_resolver;
+mod resolver;
+mod server;
+
+use std::{net::SocketAddr, sync::Arc};
+
+pub use builder::DynamicSchemaBuilder;
+pub use server::MockGraphQlServer;
+use url::Url;
+
+/// A dynamic GraphQL schema that can be built and executed at runtime.
+#[derive(Debug)]
+pub struct DynamicSchema {
```

Comment:
Would be nice to have a dedicated crate for it that we could re-use for both integration-tests & extension